### PR TITLE
Feature reports

### PR DESCRIPTION
**Pull request by WMagnus**
Adding metrics page feature, without filtering. If @carolynrehm could take a look and approve this and perhaps get the ball rolling on UAT for this, that would be wonderful.

### CODE REVIEW

File: src/test/java/com/revature/servicestests/SearchServiceTest.java [hunk 1]
```diff
@@ -62,130 +66,78 @@
...
-	@Commit
-	@Order(6)
-	void deleteTestData() {
-		JdbcTestUtils.deleteFromTableWhere(template, "module", "subject = 'FIRST TEST MODULE'");
-		JdbcTestUtils.deleteFromTableWhere(template, "module", "subject = 'SECOND TEST MODULE'");
-		JdbcTestUtils.deleteFromTableWhere(template, "content", "title = 'FIRST TEST CONTENT'");
+		assertTrue(validFilter);
+		assertFalse(badTitleFilter);
+		assertFalse(badFormatFilter);
+		assertFalse(badModuleFilter);
```
Comment (carolynrehm) [inline src/test/java/com/revature/servicestests/SearchServiceTest.java]: Each test should test one thing at a time, this appears to test 8.

File: src/main/java/com/revature/controllers/MetricsController.java [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+	
+	
+	@PostMapping("/obtain/{timeFrame}")
+	public MetricsData getMetrics(@PathVariable("timeFrame") long timeRange, 
+								  @RequestBody Map<String, Object> ids) {
+		System.out.println(ids	);
+		//formats for codeCount
+		String[] formats = new String[] {"Code", "Document", "Powerpoint"};
+		Map<String, Integer> contentFormats = contentService.getContentByFormat(formats);
+		System.out.println(contentFormats.toString());
```
Comment (carolynrehm) [inline src/main/java/com/revature/controllers/MetricsController.java]: remove sysouts or replace with logging + remove non documentation comments

File: src/main/java/com/revature/controllers/MetricsController.java [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+	@Autowired
+	ContentService contentService;
+	@Autowired
+	SearchService searchService;
+	@Autowired
+	TimegraphService timegraphService;
+		
+	
+	
+	@PostMapping("/obtain/{timeFrame}")
```
Comment (carolynrehm) [inline src/main/java/com/revature/controllers/MetricsController.java]: why obtain?

Comment (WMagnus) [reply src/main/java/com/revature/controllers/MetricsController.java]: Needs to be changed to metrics. I'll have that resolved.

File: src/test/java/com/revature/servicestests/SearchServiceTest.java (hunk 1 above)
Comment (WMagnus) [reply src/test/java/com/revature/servicestests/SearchServiceTest.java]: Tests have been split.

File: src/main/java/com/revature/controllers/MetricsController.java (hunk 2 above)
Comment (WMagnus) [reply src/main/java/com/revature/controllers/MetricsController.java]: One of my team members has handled the sysouts, and is going through looking for commented-out code.

**Review by carolynrehm — COMMENTED** [review]
Make sure tests are only testing one thing per method

File: src/test/java/com/revature/servicestests/ContentServiceTest.java [hunk 4]
```diff
@@ -18,25 +19,32 @@
 import org.springframework.test.annotation.Rollback;
 import org.springframework.test.context.ContextConfiguration;
 import org.springframework.test.context.junit.jupiter.SpringExtension;
-import org.springframework.test.jdbc.JdbcTestUtils;
-import org.springframework.transaction.annotation.Transactional;
 
 import com.revature.entities.Content;
 import com.revature.entities.Link;
+import com.revature.repositories.ContentRepository;
```
Comment (carolynrehm) [inline src/test/java/com/revature/servicestests/ContentServiceTest.java:22]: is this used?

**Comment by carolynrehm** [comment]
Still a handful of non documentation comments, remove these and you should be good to merge

File: src/test/java/com/revature/servicestests/ContentServiceTest.java (hunk 4 above)
Comment (WMagnus) [reply src/test/java/com/revature/servicestests/ContentServiceTest.java:22]: In that service, yes. It's used at a couple of points for forcible cleanup, but probably could be dropped thanks to the rollback annotation.

File: src/main/java/com/revature/controllers/MetricsController.java (hunk 2 above)
Comment (WMagnus) [reply src/main/java/com/revature/controllers/MetricsController.java]: Commentary outside of real documentation has been aggressively culled on the backend. Frontend commentary removal is pending.

**Review by carolynrehm — APPROVED** [review]
Looks good, I'll merge when front end is approved as well

**Pull request merged**